### PR TITLE
[release-0.49] Improve non-root path handling

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
Manual cherry-pick of #8198 and #8268

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
The banncheck commits were omitted because they don't seem to work with this version of Bazel.
Since banncheck is in main and all backports come from there this should be fine.
It also makes this backport less intrusive.

**Release note**:

```release-note
Improve path handling for non-root virt-launcher workloads
```

### CODE REVIEW

**Review by stu-gott — COMMENTED** [review]
/approve

Thank you!

**Review by iholder-redhat — COMMENTED** [review]
Thanks @jean-edouard! I have:
1. Compared your changes with the original PRs.
2. Tried to find paths that aren't safe, especially paths have been deleted since 4.10 and therefore are likely to be missed.

But I haven't found anything suspicious. Well done!

Left two small questions, but I don't think they should block the PR from being merged.

/lgtm

File: pkg/safepath/safepath_linux.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// +build linux
```
[the remark this answers is not in the document]
Comment (jean-edouard) [reply pkg/safepath/safepath_linux.go:1]: I think the `+` notation is just the old syntax... `//go:build linux` did not compile.

**Pull request merged**